driver/src/main/com/mongodb/client/gridfs/GridFSDownloadStreamImpl.java

```diff
@@ -41,6 +41,8 @@
     private int bufferOffset;
     private long currentPosition;
     private byte[] buffer = null;
+    private int markReadLimit;
```

Comment:
I interpret the limit as a hint to implementors of how much data they should be expected to need to cache.  Since our implementation doesn't need to cache anything (since we have an immutable file backing our stream), we don't have to cache anything.